app/helpers/authentication.rb

```diff
@@ -0,0 +1,39 @@
+def create
+  @user = User.new(user_name: params[:user_name], email: params[:email])
+  p params.inspect
+  @user.password = params[:password]
+  @user.save!
+end
+
+def login
+  @user = User.find_by(user_name: params[:user_name])
+  if !@user.is_a?(User)
+    @incorrect_login = true
+  elsif @user.password == params[:password]
+    new_session = Session.create(user_id: @user.id)
+    session[:this_id] = new_session.id
```

Comment:
`session[:this_id]` should be `session[:user_id]` or `session[:current_user_id]` `this_id` doesnt communicate the user of the session variable.